activerecord/test/cases/validations/uniqueness_validation_test.rb

```diff
@@ -386,4 +393,13 @@ def test_validate_uniqueness_on_empty_relation
     topic = TopicWithUniqEvent.new
     assert topic.valid?
   end
+
+  def test_validate_uniqueness_on_model_without_primary_key
+    MateyWithUniqValidation.create(pirate_id: 1)
+    matey = MateyWithUniqValidation.create(pirate_id: 2)
+
+    matey.decrement(:pirate_id)
```

Comment:
This is the only comment I don't agree with. Using `new` wouldn't persist the models so the uniqueness of the primary key validation would be skipped. Basically, the test would pass against the current code.